solr/modules/jwt-auth/src/java/org/apache/solr/security/jwt/JWTAuthPlugin.java

```diff
@@ -595,12 +595,26 @@ protected JWTAuthenticationResponse authenticate(String authorizationHeader) {
             } else {
               // Pull roles from separate claim, either as whitespace separated list or as JSON
               // array
-              Object rolesObj = jwtClaims.getClaimValue(rolesClaim);
+              Object rolesObj = null;
+              if (rolesClaim.indexOf('.') > 0) {
+                // support map resolution of nested values
+                String[] nestedKeys = rolesClaim.split("\\.");
+                rolesObj = jwtClaims.getClaimValue(nestedKeys[0]);
+                for (int i = 1; i < nestedKeys.length; i++) {
+                  if (rolesObj instanceof Map) {
+                    String key = nestedKeys[i];
+                    rolesObj = ((Map<?, ?>) rolesObj).get(key);
+                  }
+                }
+              } else {
+                rolesObj = jwtClaims.getClaimValue(rolesClaim);
+              }
+
               if (rolesObj != null) {
                 if (rolesObj instanceof String) {
                   finalRoles.addAll(Arrays.asList(((String) rolesObj).split("\\s+")));
                 } else if (rolesObj instanceof List) {
-                  finalRoles.addAll(jwtClaims.getStringListClaimValue(rolesClaim));
+                  ((List<?>) rolesObj).stream().forEach(entry -> finalRoles.add((String) entry));
```

Comment:
In theory the array may contain non-strings, such as objects, numeric etc. This change would make sure we only add string-values as roles:
```suggestion
                  ((List<?>) rolesObj)
                      .forEach(
                          entry -> {
                            if (entry instanceof String) {
                              finalRoles.add((String) entry);
                            }
                          });
```